contracts/DependencyRegistryV0.sol

```diff
@@ -0,0 +1,751 @@
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount;
+        require(_index < additionalRepositoryCount, "Asset index out of range");
+
+        uint24 lastElementIndex = additionalRepositoryCount - 1;
+
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+                _index
+            ] = dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+            lastElementIndex
+        ];
+        delete dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+            lastElementIndex
+        ];
+
+        dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount = lastElementIndex;
+
+        emit DependencyTypeAdditionalRepositoryRemoved(
+            _dependencyTypeId,
+            _index
+        );
+    }
+
+    /**
+     * @notice Updates additional repository for dependency type `_dependencyTypeId` at `_index`.
+     * @param _dependencyTypeId Dependency type to be updated.
+     * @param _index Additional repository index.
+     * @param _additionalRepository New Repository URL.
+     */
+    function updateDependencyTypeAdditionalRepositoryAtIndex(
+        bytes32 _dependencyTypeId,
+        uint256 _index,
+        string memory _additionalRepository
+    )
+        external
+        onlyAdminACL(
+            this.updateDependencyTypeAdditionalRepositoryAtIndex.selector
+        )
+    {
+        uint24 additionalRepositoryCount = dependencyTypeInfo[_dependencyTypeId]
+            .additionalRepositoryCount;
+        require(_index < additionalRepositoryCount, "Asset index out of range");
+
+        dependencyTypeInfo[_dependencyTypeId].additionalRepositories[
+                _index
+            ] = _additionalRepository;
+
+        emit DependencyTypeAdditionalRepositoryUpdated(
+            _dependencyTypeId,
+            _additionalRepository,
+            _index
+        );
+    }
+
+    /**
+     * @notice Returns details for depedency type `_dependencyTypeId`.
+     * @param _dependencyTypeId Dependency type to be queried.
+     * @return typeAndVersion String representation of `_dependencyTypeId`.
+     * (e.g. "p5js(atSymbol)1.0.0")
+     * @return preferredCDN Preferred CDN URL for dependency type
+     * @return additionalCDNCount Count of additional CDN URLs for dependency type
+     * @return preferredRepository Preferred repository URL for dependency type
+     * @return additionalRepositoryCount Count of additional repository URLs for dependency type
+     * @return projectWebsite Project website URL for dependency type
+     * @return availableOnChain Whether dependency type is available on chain
+     * @return scriptCount Count of on-chain scripts for dependency type
+     */
+    function dependencyTypeDetails(bytes32 _dependencyTypeId)
+        external
+        view
+        returns (
+            string memory typeAndVersion,
```

Comment:
do you think we should return this, if it is just the string version of what the user is already sending as calldata?

Reply:
It seems most idiomatic to include it in response to allow for response being display-able without client (e.g. imagine a "registry explorer" client app) needing to combine back together the call data and the response client-side.